ftd/src/ast/component.rs

```diff
@@ -513,29 +601,14 @@ impl Loop {
             );
         }
 
-        let alias = {
-            if let Some(alias) = alias {
-                if !alias.starts_with(ftd::ast::utils::REFERENCE) {
-                    return ftd::ast::parse_error(
-                        format!(
-                            "Loop alias should start with reference, found: `{}`. Help: use `${}` instead",
-                            alias, alias
-                        ),
-                        doc_id,
-                        loop_header.get_line_number(),
-                    );
-                }
-                alias
-                    .trim_start_matches(ftd::ast::utils::REFERENCE)
-                    .to_string()
-            } else {
-                "object".to_string()
-            }
-        };
+        let alias = alias
+            .trim_start_matches(ftd::ast::utils::REFERENCE)
+            .to_string();
 
         Ok(Some(Loop::new(
             on.as_str(),
             alias.as_str(),
+            key_name,
```

Comment:
Rename key_name to loop_counter_alias
```suggestion
            loop_counter_alias,
```